code/datums/brain_damage/severe.dm

```diff
@@ -136,25 +136,61 @@
 	scan_desc = "traumatic narcolepsy"
 	gain_text = span_warning("You have a constant feeling of drowsiness...")
 	lose_text = span_notice("You feel awake and aware again.")
+	/// Odds seconds_per_tick the user falls asleep
+	var/sleep_chance = 1
+	/// Odds seconds_per_tick the user falls asleep while running
+	var/sleep_chance_running = 2
+	/// Odds seconds_per_tick the user falls asleep while drowsy
+	var/sleep_chance_drowsy = 3
+	/// Time values for how long the user will stay drowsy
+	var/drowsy_time_minimum = 20 SECONDS
+	var/drowsy_time_maximum = 30 SECONDS
+	/// Time values for how long the user will stay asleep
+	var/sleep_time_minimum = 3 SECONDS
+	var/sleep_time_maximum = 6 SECONDS
 
 /datum/brain_trauma/severe/narcolepsy/on_life(seconds_per_tick, times_fired)
 	if(owner.IsSleeping())
 		return
 
-	var/sleep_chance = 1
+	/// If any of these are in the user's blood, return early
+	var/list/immunity_medicine = list(
+		/datum/reagent/medicine/modafinil,
+		/datum/reagent/medicine/synaptizine,
+	) //don't add too many, as most stimulant reagents already have a drowsy-removing effect
+	for(var/datum/reagent/medicine in immunity_medicine)
+		if(owner.reagents.has_reagent(medicine))
+			return
```

Comment:
Also: This code will not function

It's type checking for `/datum/reagent` but the list only has typeaths, not datums


```suggestion
	for(var/datum/reagent/medicine as anything in immunity_medicine)
		if(owner.reagents.has_reagent(medicine))
			return
```